tokio-macros/src/entry.rs

```diff
@@ -1,11 +1,171 @@
+        &mut self,
+        num_workers: syn::Lit,
+        span: Span,
+    ) -> Result<(), syn::Error> {
+        if self.num_workers.is_some() {
+            return Err(syn::Error::new(span, "`num_workers` set multiple times."));
+        }
+
+        let num_workers = parse_int(num_workers, span, "num_workers")?;
+        if num_workers == 0 {
+            return Err(syn::Error::new(span, "`num_workers` may not be 0."));
+        }
+        self.num_workers = Some((num_workers, span));
+        Ok(())
+    }
+    fn set_max_threads(
+        &mut self,
+        max_threads: syn::Lit,
+        span: Span,
+    ) -> Result<(), syn::Error> {
+        if self.max_threads.is_some() {
+            return Err(syn::Error::new(span, "`max_threads` set multiple times."));
+        }
+
+        let max_threads = parse_int(max_threads, span, "max_threads")?;
+        if max_threads == 0 {
+            return Err(syn::Error::new(span, "`max_threads` may not be 0."));
+        }
+        self.max_threads = Some((max_threads, span));
+        Ok(())
+    }
+    fn build(&self) -> Result<FinalConfig, syn::Error> {
+        let flavor = self.flavor.unwrap_or(self.default_flavor);
+        match flavor {
+            RuntimeFlavor::InPlace => {
+                if let Some((_, num_workers_span)) = &self.num_workers {
+                    return Err(syn::Error::new(
+                        num_workers_span.clone(),
+                        "The `num_workers` option requires the `work_stealing` runtime flavor."
+                    ));
+                }
+                assert!(self.num_workers.is_none());
+                Ok(FinalConfig {
+                    flavor,
+                    num_workers: None,
+                    max_threads: self.max_threads.map(|(val, _span)| val),
+                })
+            },
+            RuntimeFlavor::WorkStealing => {
+                if !self.rt_threaded_available {
+                    let msg = if self.flavor.is_none() {
+                        "The default runtime flavor is `work_stealing`, but the `rt-threaded` feature is disabled."
+                    } else {
+                        "The runtime flavor `work_stealing` requires the `rt-threaded` feature."
+                    };
+                    return Err(syn::Error::new(Span::call_site(), msg));
+                }
+                match (self.num_workers, self.max_threads) {
+                    (None, Some((_, span))) => {
+                        return Err(syn::Error::new(
+                                span,
+                                "`num_workers` must also be set when using `max_threads`"
+                        ));
+                    },
+                    (Some((num_workers, _)), Some((max_threads, span))) if max_threads <= num_workers => {
+                        return Err(syn::Error::new(
+                                span,
+                                "`max_threads` must be larger than `num_workers`",
```

Comment:
In this case, can we just default to `max(core_threads, max_threads)`? I think that is a better default and I think we should do the same in the builder.

Reply:
Allowing `max_threads == core_threads` seems dangerous and like an edge case that I'm not sure we should support from the macro.

Reply:
we could do `max_threads = core_threads + 1` at a min. That said, lets not do this here since I think this could also be a potential footgun. Honestly, should we even allow someone to set max threads here?